libvast/vast/table_slice_column.hpp

```diff
@@ -14,27 +14,64 @@
 #pragma once
 
 #include "vast/fwd.hpp"
-#include "vast/table_slice.hpp"
+#include "vast/view.hpp"
+
+#include <caf/meta/type_name.hpp>
 
 #include <cstdint>
+#include <string_view>
 
 namespace vast {
 
-struct table_slice_column {
-  table_slice_column() {
-  }
+/// Convenience helper for traversing a column of a table slice.
+/// @relates table_slice
+class table_slice_column {
+public:
+  /// Defaulted constructors, destructors, and assignment operators.
+  table_slice_column() noexcept;
+  ~table_slice_column() noexcept;
+  table_slice_column(const table_slice_column&) noexcept;
+  table_slice_column& operator=(const table_slice_column&) noexcept;
+  table_slice_column(table_slice_column&&) noexcept;
+  table_slice_column& operator=(table_slice_column&&) noexcept;
 
-  table_slice_column(table_slice_ptr slice, size_t col)
-    : slice{std::move(slice)}, column{col} {
-    // nop
-  }
-  table_slice_ptr slice;
-  size_t column;
+  /// Construct a view on a column of a table slice.
+  /// @param slice The slice to view.
+  /// @param column The viewed column's index.
+  /// @pre `column < slice->columns()`
+  table_slice_column(table_slice_ptr slice, size_t column) noexcept;
```

Comment:
I think both of those constructors should be `private`. Instead, we should provide
```
static std::optional<table_slice_column>(table_slice_ptr slice, size_t column);
static std::optional<table_slice_column>(table_slice_ptr slice, std::string_view column);
```
Optional or pointer semantics should not creep into the actual `table_slice_column`.

Reply:
I'd argue this only really makes sense for the latter constructor. Will change for that one.

Reply:
As long as we keep the precondition it is ok.